src/network/networkd-netdev-wireguard.c

```diff
@@ -0,0 +1,625 @@
+                                      const char *rvalue,
+                                      void *data,
+                                      void *userdata) {
+        Wireguard *w;
+        WireguardPeer *peer;
+
+        assert(data);
+
+        w = WIREGUARD(data);
+
+        assert(w);
+
+        peer = new_peer(w, section_line);
+        if (!peer)
+                return -ENOMEM;
+
+        return parse_wireguard_key(unit,
+                                   filename,
+                                   line,
+                                   section,
+                                   section_line,
+                                   lvalue,
+                                   ltype,
+                                   rvalue,
+                                   data,
+                                   peer->fields.public_key);
+}
+
+int config_parse_wireguard_allowed_ips(const char *unit,
+                                       const char *filename,
+                                       unsigned line,
+                                       const char *section,
+                                       unsigned section_line,
+                                       const char *lvalue,
+                                       int ltype,
+                                       const char *rvalue,
+                                       void *data,
+                                       void *userdata) {
+        _cleanup_strv_free_ char **words = NULL;
+        char **word;
+        const char *address, *end;
+        union in_addr_union buffer;
+        unsigned char prefixlen;
+        int r, family;
+        Wireguard *w;
+        WireguardPeer *peer;
+        WireguardIpmask *ipmask;
+
+        assert(rvalue);
+        assert(data);
+
+        w = WIREGUARD(data);
+
+        peer = new_peer(w, section_line);
+        if (!peer)
+                return -ENOMEM;
+
+        r = strv_split_extract(&words, rvalue, ",", 0);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to split allowed ips \"%s\" option: %m", rvalue);
+                return 0;
+        }
+        STRV_FOREACH(word, words) {
+                end = strchr(*word, '/');
+                if (end)
+                        address = strndupa(*word, end - *word);
+                else
+                        address = *word;
+
+                r = in_addr_from_string_auto(address, &family, &buffer);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Ip is invalid, ignoring assignment: %s", *word);
+                        return 0;
+                }
+                if (end) {
+                        r = safe_atou8(end + 1, &prefixlen);
+                        if (r < 0) {
+                                log_syntax(unit, LOG_ERR, filename, line, r, "Route destination prefix length is invalid, ignoring assignment: %s", end + 1);
+                                return 0;
```

Comment:
@poettering I thought about parsing both, the address and subnet size in one function from CIDR notation (see `cidr_from_string`). This should be more common than a standalone prefix, no? There is one other instance in networkd, where this can be reused: networkd-route.c:722